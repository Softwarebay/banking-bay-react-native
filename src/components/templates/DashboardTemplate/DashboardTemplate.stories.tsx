import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { CenterStory } from 'utils/storybook/CenterStory';

import { DashboardTemplate, TransactionHistoryItem } from './DashboardTemplate';

const latestTransactions: TransactionHistoryItem[] = [
	{
		title: 'Adalyn Roth',
		id: 'adalyn-roth-124743782',
		iconColor: '#55ACEE',
		amount: 140.0,
		iconName: 'money-transfer',
		transactionCategory: 'Money transfer',
		transactionType: 'outgoing',
	},
	{
		title: 'Amazon',
		id: 'amazon-124743782',
		iconColor: 'none',
		amount: 239.57,
		iconName: 'amazon',
		transactionCategory: 'Online payments',
		transactionType: 'outgoing',
	},
	{
		title: 'Paypal',
		id: 'paypal-124743782',
		iconColor: 'none',
		amount: 700.0,
		iconName: 'paypal',
		transactionCategory: 'Deposits',
		transactionType: 'incoming',
	},
];

storiesOf('Templates/DashboardTemplate', module)
	.addDecorator((getStory) => (
		<CenterStory style={{ paddingHorizontal: 0 }}>{getStory()}</CenterStory>
	))
	.add('Default', () => (
		<DashboardTemplate
			latestTransactions={latestTransactions}
			navText={text('navText', 'navText')}
			onPressCreditCardsIconButton={action('onPressCreditCardsIconButton')}
			onPressHeaderLink={action('onPressHeaderLink')}
			onPressMakePayment={action('onPressMakePayment')}
			onPressMobilePayment={action('onPressMobilePayment')}
			onPressMoneyTransfer={action('onPressMoneyTransfer')}
			onPressProfileIconButton={action('onPressProfileIconButton')}
			onPressTopUpPayment={action('onPressTopUpPayment')}
		/>
	));
