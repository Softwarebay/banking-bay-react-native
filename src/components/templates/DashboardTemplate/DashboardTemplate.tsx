import { StatusBar } from 'expo-status-bar';
import { View } from 'react-native';

import { Divider, IconProps } from 'components/atoms';
import {
	DashboardCards,
	LatestTransactions,
	PaymentOptions,
} from 'components/organisms';

import { styles } from './DashboardTemplate.styles';

export interface TransactionHistoryItem {
	id: string;
	iconName: IconProps['name'];
	iconColor?: string;
	transactionType: 'incoming' | 'outgoing';
	title: string;
	transactionCategory: string;
	amount: number;
}

export interface Props {
	navText: string;
	latestTransactions: TransactionHistoryItem[];
	onPressCreditCardsIconButton: () => void;
	onPressProfileIconButton: () => void;
	onPressMakePayment: () => void;
	onPressTopUpPayment: () => void;
	onPressMobilePayment: () => void;
	onPressMoneyTransfer: () => void;
	onPressViewAll: () => void;
	onPressTransactionCard: () => void;
}

export const DashboardTemplate = ({
	navText,
	latestTransactions,
	onPressCreditCardsIconButton,
	onPressProfileIconButton,
	onPressMakePayment,
	onPressTopUpPayment,
	onPressMobilePayment,
	onPressMoneyTransfer,
	onPressViewAll,
	onPressTransactionCard,
}: Props) => {
	return (
		<View style={styles.container}>
			<StatusBar style="light" />
			<DashboardCards
				navText={navText}
				onPressCreditCardsIconButton={onPressCreditCardsIconButton}
				onPressProfileIconButton={onPressProfileIconButton}
			/>
			<View style={styles.content}>
				<View style={styles.paymentOptions}>
					<PaymentOptions
						onPressMakePayment={onPressMakePayment}
						onPressMobilePayment={onPressMobilePayment}
						onPressMoneyTransfer={onPressMoneyTransfer}
						onPressTopUpPayment={onPressTopUpPayment}
					/>
				</View>
				<Divider testID="dashboard-template-divider-test-id" />
				<LatestTransactions
					latestTransactions={latestTransactions}
					style={styles.latestTransactions}
					onPressTransactionCard={onPressTransactionCard}
					onPressViewAll={onPressViewAll}
				/>
			</View>
		</View>
	);
};
