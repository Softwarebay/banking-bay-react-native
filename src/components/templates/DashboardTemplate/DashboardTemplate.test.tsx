import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { DashboardTemplate } from './DashboardTemplate';

import type { Props } from './DashboardTemplate';

describe('DashboardTemplate', () => {
	const onPressProfileIconButton = jest.fn<
		Props['onPressProfileIconButton'],
		[]
	>();
	const onPressCreditCardsIconButton = jest.fn<
		Props['onPressCreditCardsIconButton'],
		[]
	>();
	const onPressMakePayment = jest.fn<Props['onPressMakePayment'], []>();
	const onPressMobilePayment = jest.fn<Props['onPressMobilePayment'], []>();
	const onPressMoneyTransfer = jest.fn<Props['onPressMoneyTransfer'], []>();
	const onPressTopUpPayment = jest.fn<Props['onPressTopUpPayment'], []>();
	const onPressTransactionCard = jest.fn<Props['onPressTransactionCard'], []>();
	const onPressViewAll = jest.fn<Props['onPressViewAll'], []>();

	const props: Props = {
		latestTransactions: [
			{
				id: 'item1',
				amount: 2.0,
				iconName: 'amazon',
				title: 'Item1',
				transactionCategory: 'Mobile',
				transactionType: 'outgoing',
			},
			{
				id: 'item2',
				amount: 3.0,
				iconName: 'amazon',
				title: 'Item2',
				transactionCategory: 'Mobile',
				transactionType: 'outgoing',
			},
		],
		navText: 'Dashboard Nav',
		onPressProfileIconButton,
		onPressCreditCardsIconButton,
		onPressMakePayment,
		onPressMobilePayment,
		onPressMoneyTransfer,
		onPressTopUpPayment,
		onPressTransactionCard,
		onPressViewAll,
	};

	it('should match snapshot', () => {
		const tree = testRenderer.create(<DashboardTemplate {...props} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render dashboard header with nav text, profile icon button, credit cards icon button and 3 card images', () => {
		render(<DashboardTemplate {...props} />);

		const navText = screen.getByTestId('dashboard-nav-text-test-id');
		const profileIcon = screen.getByTestId('profile-icon-test-id');
		const cardsIcon = screen.getByTestId('credit-cards-icon-test-id');
		const cardImgs = screen.getAllByTestId(/(card1|card2)-img-test-id/);

		expect(navText).toHaveTextContent('Dashboard Nav');
		expect(profileIcon).not.toBeNull();
		expect(cardsIcon).not.toBeNull();
		expect(cardImgs).toHaveLength(3);
	});

	it('should handle press events on header icon buttons', () => {
		render(<DashboardTemplate {...props} />);

		const cardsBtn = screen.getByTestId('dashboard-cards-btn-test-id');
		const profileBtn = screen.getByTestId('dashboard-profile-icon-btn-test-id');

		fireEvent.press(cardsBtn);
		expect(onPressCreditCardsIconButton).toHaveBeenCalledTimes(1);

		fireEvent.press(profileBtn);
		expect(onPressProfileIconButton).toHaveBeenCalledTimes(1);
	});

	it('should render 4 payment option buttons', () => {
		render(<DashboardTemplate {...props} />);

		const paymentBtns = screen.getAllByTestId(
			/(top-up-payment|mobile-payment|money-transfer|make-payment)-icon-btn-test-id/gi
		);

		expect(paymentBtns).toHaveLength(4);
	});

	it("should handle press events on 'Make a Payment' icon button", () => {
		render(<DashboardTemplate {...props} />);

		const makePaymentIconBtn = screen.getByTestId(
			'make-payment-icon-btn-test-id'
		);

		fireEvent.press(makePaymentIconBtn);

		expect(onPressMakePayment).toHaveBeenCalledTimes(1);
		expect(onPressMobilePayment).not.toHaveBeenCalled();
		expect(onPressMoneyTransfer).not.toHaveBeenCalled();
		expect(onPressTopUpPayment).not.toHaveBeenCalled();
	});

	it("should handle press events on 'Mobile Payment' icon button", () => {
		render(<DashboardTemplate {...props} />);

		const mobilePaymentIconBtn = screen.getByTestId(
			'mobile-payment-icon-btn-test-id'
		);

		fireEvent.press(mobilePaymentIconBtn);

		expect(onPressMobilePayment).toHaveBeenCalledTimes(1);
		expect(onPressMakePayment).not.toHaveBeenCalled();
		expect(onPressMoneyTransfer).not.toHaveBeenCalled();
		expect(onPressTopUpPayment).not.toHaveBeenCalled();
	});

	it("should handle press events on 'Money Transfer' icon button", () => {
		render(<DashboardTemplate {...props} />);

		const moneyTransferIconBtn = screen.getByTestId(
			'money-transfer-icon-btn-test-id'
		);

		fireEvent.press(moneyTransferIconBtn);

		expect(onPressMoneyTransfer).toHaveBeenCalledTimes(1);
		expect(onPressMakePayment).not.toHaveBeenCalled();
		expect(onPressMobilePayment).not.toHaveBeenCalled();
		expect(onPressTopUpPayment).not.toHaveBeenCalled();
	});

	it("should handle press events on 'Top-Up Payment' icon button", () => {
		render(<DashboardTemplate {...props} />);

		const topUpPaymentIconBtn = screen.getByTestId(
			'top-up-payment-icon-btn-test-id'
		);

		fireEvent.press(topUpPaymentIconBtn);

		expect(onPressTopUpPayment).toHaveBeenCalledTimes(1);
		expect(onPressMakePayment).not.toHaveBeenCalled();
		expect(onPressMobilePayment).not.toHaveBeenCalled();
		expect(onPressMoneyTransfer).not.toHaveBeenCalled();
	});

	it('should render divider', () => {
		render(<DashboardTemplate {...props} />);

		const divider = screen.getByTestId('dashboard-template-divider-test-id');

		expect(divider).not.toBeNull();
	});

	it("should render 'Latest transaction' section with a title and 'View all' button", () => {
		render(<DashboardTemplate {...props} />);

		const title = screen.getByTestId('latest-transactions-title-test-id');
		const btn = screen.getByTestId('latest-transaction-link-test-id');

		expect(title).toHaveTextContent('Latest transactions');
		expect(btn).toHaveTextContent('View All');
	});

	it("should handle press events on 'View all' button", () => {
		render(<DashboardTemplate {...props} />);

		const btn = screen.getByTestId('latest-transaction-link-test-id');

		fireEvent.press(btn);

		expect(onPressViewAll).toHaveBeenCalledTimes(1);
	});

	it('should render latest transactions as a list of transaction cards', () => {
		render(<DashboardTemplate {...props} />);

		const transactionItems = screen.getAllByTestId(
			/-latest-transaction-item-test-id/
		);

		expect(transactionItems).toHaveLength(2);
	});
	it('should handle press events on transaction card', () => {
		render(<DashboardTemplate {...props} />);

		const transactionItems = screen.getAllByTestId(
			/-latest-transaction-item-test-id/
		);

		transactionItems.forEach((item) => {
			fireEvent.press(item);
		});

		expect(onPressTransactionCard).toHaveBeenCalledTimes(2);
	});
});
