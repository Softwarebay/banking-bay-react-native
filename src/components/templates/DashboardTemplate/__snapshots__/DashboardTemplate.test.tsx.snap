// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardTemplate should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#1B1D4D",
        "borderBottomLeftRadius": 30,
        "borderBottomRightRadius": 30,
        "flex": 0.55,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "width": 750,
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": 750,
            },
            undefined,
          ]
        }
      />
      <RNCSafeAreaView
        edges={
          Array [
            "bottom",
            "left",
            "right",
            "top",
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
              },
              Object {
                "flexDirection": "row",
              },
              Object {
                "alignItems": "center",
                "justifyContent": "space-between",
                "paddingHorizontal": 20,
                "paddingTop": 20,
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={20}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                undefined,
                false,
              ]
            }
            testID="dashboard-profile-icon-btn-test-id"
          >
            <SvgMock
              color="#3B5999"
              height={25}
              testID="profile-icon-test-id"
              width={25}
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#4C4C60",
                  "fontFamily": "MulishRegular",
                  "fontSize": 17,
                  "lineHeight": 27.2,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
            testID="dashboard-nav-text-test-id"
          >
            Dashboard Nav
          </Text>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={20}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                undefined,
                false,
              ]
            }
            testID="dashboard-cards-btn-test-id"
          >
            <SvgMock
              color="#FFFFFF"
              height={25}
              testID="credit-cards-icon-test-id"
              width={25}
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "#CED6E1",
                "borderTopWidth": 1,
              },
              Object {
                "marginVertical": 20,
                "opacity": 0.3,
              },
            ]
          }
          testID="dashboards-cards-divider-test-id"
        />
        <RCTScrollView
          contentContainerStyle={
            Object {
              "flexDirection": "row",
            }
          }
          data={
            Array [
              Object {
                "id": 1,
                "img": "card1",
              },
              Object {
                "id": 2,
                "img": "card2",
              },
              Object {
                "id": 3,
                "img": "card1",
              },
            ]
          }
          decelerationRate="fast"
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={true}
          initialScrollIndex={0}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          pagingEnabled={true}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          showsHorizontalScrollIndicator={false}
          snapToAlignment="center"
          stickyHeaderIndices={Array []}
          viewabilityConfigCallbackPairs={
            Array [
              Object {
                "onViewableItemsChanged": [Function],
                "viewabilityConfig": Object {
                  "viewAreaCoveragePercentThreshold": 50,
                },
              },
            ]
          }
        >
          <View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <Image
                source={1}
                style={
                  Array [
                    Object {
                      "height": undefined,
                    },
                    Object {
                      "aspectRatio": 1.5472312703583062,
                    },
                    Object {
                      "width": 750,
                    },
                  ]
                }
                testID="card1-img-test-id"
              />
            </View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <Image
                source={1}
                style={
                  Array [
                    Object {
                      "height": undefined,
                    },
                    Object {
                      "aspectRatio": 1.5472312703583062,
                    },
                    Object {
                      "width": 750,
                    },
                  ]
                }
                testID="card2-img-test-id"
              />
            </View>
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  null,
                ]
              }
            >
              <Image
                source={1}
                style={
                  Array [
                    Object {
                      "height": undefined,
                    },
                    Object {
                      "aspectRatio": 1.5472312703583062,
                    },
                    Object {
                      "width": 750,
                    },
                  ]
                }
                testID="card1-img-test-id"
              />
            </View>
          </View>
        </RCTScrollView>
        <View
          style={
            Object {
              "alignItems": "center",
              "marginTop": -10,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                },
                Object {
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
            testID="dashboard-cards-paginator-test-id"
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#1B1D4D",
                    "borderRadius": 100,
                    "borderWidth": 3,
                    "height": 8,
                    "marginHorizontal": 8,
                    "opacity": 0.2,
                    "width": 8,
                  },
                  Object {
                    "opacity": 1,
                  },
                  Object {
                    "borderColor": "#FFFFFF",
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#1B1D4D",
                    "borderRadius": 100,
                    "borderWidth": 3,
                    "height": 8,
                    "marginHorizontal": 8,
                    "opacity": 0.2,
                    "width": 8,
                  },
                  false,
                  Object {
                    "borderColor": "#FFFFFF",
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#1B1D4D",
                    "borderRadius": 100,
                    "borderWidth": 3,
                    "height": 8,
                    "marginHorizontal": 8,
                    "opacity": 0.2,
                    "width": 8,
                  },
                  false,
                  Object {
                    "borderColor": "#FFFFFF",
                  },
                ]
              }
            />
          </View>
        </View>
      </RNCSafeAreaView>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 0.5,
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": -15,
          "transform": Array [
            Object {
              "translateY": -30,
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-around",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "height": "100%",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={20}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Array [
                  Object {
                    "borderRadius": 100,
                    "height": 70,
                    "marginBottom": 10,
                    "width": 70,
                  },
                  Object {
                    "backgroundColor": "#3EB290",
                  },
                ],
                false,
              ]
            }
            testID="top-up-payment-icon-btn-test-id"
          >
            <SvgMock
              color="#FFFFFF"
              height={25}
              testID="credit-card-icon-test-id"
              width={25}
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#4C4C60",
                  "fontFamily": "MulishRegular",
                  "fontSize": 17,
                  "lineHeight": 27.2,
                },
                Object {
                  "fontFamily": "MulishSemiBold",
                  "fontSize": 12,
                  "lineHeight": 14.4,
                  "maxWidth": 70,
                  "textAlign": "center",
                },
              ]
            }
            testID="top-up-payment-title-test-id"
          >
            Top-Up Payment
          </Text>
        </View>
        <View
          style={
            Object {
              "height": "100%",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={20}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Array [
                  Object {
                    "borderRadius": 100,
                    "height": 70,
                    "marginBottom": 10,
                    "width": 70,
                  },
                  Object {
                    "backgroundColor": "#FF8A71",
                  },
                ],
                false,
              ]
            }
            testID="mobile-payment-icon-btn-test-id"
          >
            <SvgMock
              color="#FFFFFF"
              height={25}
              testID="smartphone-icon-test-id"
              width={25}
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#4C4C60",
                  "fontFamily": "MulishRegular",
                  "fontSize": 17,
                  "lineHeight": 27.2,
                },
                Object {
                  "fontFamily": "MulishSemiBold",
                  "fontSize": 12,
                  "lineHeight": 14.4,
                  "maxWidth": 70,
                  "textAlign": "center",
                },
              ]
            }
            testID="mobile-payment-title-test-id"
          >
            Mobile Payment
          </Text>
        </View>
        <View
          style={
            Object {
              "height": "100%",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={20}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Array [
                  Object {
                    "borderRadius": 100,
                    "height": 70,
                    "marginBottom": 10,
                    "width": 70,
                  },
                  Object {
                    "backgroundColor": "#55ACEE",
                  },
                ],
                false,
              ]
            }
            testID="money-transfer-icon-btn-test-id"
          >
            <SvgMock
              color="#FFFFFF"
              height={25}
              testID="repeat-icon-test-id"
              width={25}
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#4C4C60",
                  "fontFamily": "MulishRegular",
                  "fontSize": 17,
                  "lineHeight": 27.2,
                },
                Object {
                  "fontFamily": "MulishSemiBold",
                  "fontSize": 12,
                  "lineHeight": 14.4,
                  "maxWidth": 70,
                  "textAlign": "center",
                },
              ]
            }
            testID="money-transfer-title-test-id"
          >
            Money Transfer
          </Text>
        </View>
        <View
          style={
            Object {
              "height": "100%",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={20}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Array [
                  Object {
                    "borderRadius": 100,
                    "height": 70,
                    "marginBottom": 10,
                    "width": 70,
                  },
                  Object {
                    "backgroundColor": "#EECC55",
                  },
                ],
                false,
              ]
            }
            testID="make-payment-icon-btn-test-id"
          >
            <SvgMock
              color="#FFFFFF"
              height={25}
              testID="dollar-sign-icon-test-id"
              width={25}
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#4C4C60",
                  "fontFamily": "MulishRegular",
                  "fontSize": 17,
                  "lineHeight": 27.2,
                },
                Object {
                  "fontFamily": "MulishSemiBold",
                  "fontSize": 12,
                  "lineHeight": 14.4,
                  "maxWidth": 70,
                  "textAlign": "center",
                },
              ]
            }
            testID="make-payment-title-test-id"
          >
            Make a Payment
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#CED6E1",
            "borderTopWidth": 1,
          },
          undefined,
        ]
      }
      testID="dashboard-template-divider-test-id"
    />
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 30,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Object {
                "fontFamily": "MulishMedium",
                "fontSize": 22,
                "lineHeight": 26.4,
              },
              undefined,
            ]
          }
          testID="latest-transactions-title-test-id"
        >
          Latest transactions
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={20}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={false}
          testID="latest-transaction-link-test-id"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FF5887",
                  "fontFamily": "MulishMedium",
                  "fontSize": 17,
                  "lineHeight": 27.2,
                },
                undefined,
              ]
            }
          >
            View All
          </Text>
        </View>
      </View>
      <RCTScrollView
        data={
          Array [
            Object {
              "amount": 2,
              "iconName": "amazon",
              "id": "item1",
              "title": "Item1",
              "transactionCategory": "Mobile",
              "transactionType": "outgoing",
            },
            Object {
              "amount": 3,
              "iconName": "amazon",
              "id": "item2",
              "title": "Item2",
              "transactionCategory": "Mobile",
              "transactionType": "outgoing",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              testID="item1-latest-transaction-item-test-id"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                    },
                    Object {
                      "flexDirection": "row",
                    },
                    Array [
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 10,
                        "shadowColor": "#161966",
                        "shadowOffset": Object {
                          "height": 6,
                          "width": 0,
                        },
                        "shadowOpacity": 0.08,
                        "shadowRadius": 16,
                      },
                      Array [
                        Object {
                          "paddingHorizontal": 15,
                          "paddingVertical": 15,
                        },
                        Array [
                          Object {
                            "marginBottom": 10,
                          },
                          Object {
                            "marginTop": 20,
                          },
                          false,
                        ],
                        false,
                      ],
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#F1F5FD",
                      "borderRadius": 100,
                      "height": 50,
                      "justifyContent": "center",
                      "width": 50,
                    }
                  }
                >
                  <SvgMock
                    height={25}
                    testID="amazon-icon-test-id"
                    width={25}
                  />
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "marginHorizontal": 15,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#1B1D4D",
                        },
                        Object {
                          "fontFamily": "MulishSemiBold",
                          "fontSize": 16,
                          "lineHeight": 25.6,
                        },
                        undefined,
                      ]
                    }
                    testID="item1-transaction-title-test-id"
                  >
                    Item1
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#4C4C60",
                          "fontFamily": "MulishRegular",
                          "fontSize": 17,
                          "lineHeight": 27.2,
                        },
                        Object {
                          "fontSize": 14,
                          "lineHeight": 22.4,
                        },
                      ]
                    }
                    testID="item1-transaction-subtitle-test-id"
                  >
                    Mobile
                  </Text>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#1B1D4D",
                      },
                      Object {
                        "fontFamily": "MulishSemiBold",
                        "fontSize": 16,
                        "lineHeight": 25.6,
                      },
                      undefined,
                    ]
                  }
                  testID="item1-transaction-amount-test-id"
                >
                  - 2.00
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              testID="item2-latest-transaction-item-test-id"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                    },
                    Object {
                      "flexDirection": "row",
                    },
                    Array [
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 10,
                        "shadowColor": "#161966",
                        "shadowOffset": Object {
                          "height": 6,
                          "width": 0,
                        },
                        "shadowOpacity": 0.08,
                        "shadowRadius": 16,
                      },
                      Array [
                        Object {
                          "paddingHorizontal": 15,
                          "paddingVertical": 15,
                        },
                        Array [
                          Object {
                            "marginBottom": 10,
                          },
                          false,
                          Object {
                            "marginBottom": 20,
                          },
                        ],
                        false,
                      ],
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#F1F5FD",
                      "borderRadius": 100,
                      "height": 50,
                      "justifyContent": "center",
                      "width": 50,
                    }
                  }
                >
                  <SvgMock
                    height={25}
                    testID="amazon-icon-test-id"
                    width={25}
                  />
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "marginHorizontal": 15,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#1B1D4D",
                        },
                        Object {
                          "fontFamily": "MulishSemiBold",
                          "fontSize": 16,
                          "lineHeight": 25.6,
                        },
                        undefined,
                      ]
                    }
                    testID="item2-transaction-title-test-id"
                  >
                    Item2
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#4C4C60",
                          "fontFamily": "MulishRegular",
                          "fontSize": 17,
                          "lineHeight": 27.2,
                        },
                        Object {
                          "fontSize": 14,
                          "lineHeight": 22.4,
                        },
                      ]
                    }
                    testID="item2-transaction-subtitle-test-id"
                  >
                    Mobile
                  </Text>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#1B1D4D",
                      },
                      Object {
                        "fontFamily": "MulishSemiBold",
                        "fontSize": 16,
                        "lineHeight": 25.6,
                      },
                      undefined,
                    ]
                  }
                  testID="item2-transaction-amount-test-id"
                >
                  - 3.00
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
